src/Microsoft.ApiDesignGuidelines.Analyzers/Core/NonConstantFieldsShouldNotBeVisible.cs

```diff
@@ -32,8 +33,23 @@ public abstract class NonConstantFieldsShouldNotBeVisibleAnalyzer : DiagnosticAn
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
 
         public override void Initialize(AnalysisContext analysisContext)
-        { 
-            
+        {
+            analysisContext.RegisterSymbolAction(obj =>
+            {
+                var field = (IFieldSymbol)obj.Symbol;
+
+                if (!field.IsPublic())
+                {
+                    return;
+                }
+
+                if (field.IsConst || (field.IsStatic && field.IsReadOnly))
+                {
+                    return;
+                }
+
+                obj.ReportDiagnostic(field.CreateDiagnostic(Rule));
```

Comment:
The rule for instance fields is CA1051: Do not declare visible instance fields. So, yes, this should only check statics. Maybe they should be combined in to the same analyzer?